teps/0138-decouple-api-and-feature-versioning.md

```diff
@@ -72,8 +91,331 @@ Per [TEP-0033](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-
 **Tekton Maintainers**
 - I would like to be able to migrate the apiVersion without having to make backwards incompatible changes.
 
+## Requirements
+- Avoid making changes to existing features' stability levels just for the purpose of addressing coupling feature and API versioning.
+- Avoid blocking the promotion of `beta` features from the existing `alpha` features.
+- It should have a testing stratey that will give us confidence in our impelmentations of per feature flags and changes to existing feature flags.
+
+## Proposal
+
+This TEP provides a plan for ensuring that feature stability doesn't depend on CRD API version, and using per-feature flags to move to a future where only stable features are enabled by default through the following steps:
+- [Change existing validation to decouple feature and API versioning](#change-existing-validation-to-decouple-feature-and-api-versioning)
+- [Per feature flag for new api-driven features](#per-feature-flag-for-new-api-driven-features)
+- [Sunset `enable-api-fields` after existing features stabilize](#sunset-enable-api-fields-after-existing-features-stabilize)
+
+
+## Design Details
+
+### Change existing validation to decouple feature and API versioning
+We will change the current validation for `enable-api-fields=stable` to only allow using stable features regardless of API version. This will resolve the current issue of the coupling of API and feature versioning in v1beta1. More specifically, beta features resolvers, object/ array params and results will require `enable-api-fields` set to `beta` to be used. This means that, for those current beta features (resolvers, object/ array params and results) will no longer be enabled with `enable-api-fields=stable`. 
+To continue using these features, users will need to explicitly set the `enable-api-fields` flag to `beta`. This change will not affect users who are already using the `enable-api-fields=beta` flag, which is the default and will continue to be. It would change the behaviour for those who only want to enable `stable` features. 
+
+Note that although this is a behavior change, it is more of a bug fix for the coupling of feature and API versioning. Currently, with `enable-api-fields` set to `stable`,  PipelineRuns like [this one](https://github.com/tektoncd/pipeline/blob/main/examples/v1/pipelineruns/beta/git-resolver.yaml) fail because the controller cannot create child TaskRuns. This change will result in a validation failure instead.
```

Comment:
```suggestion
Note that although this looks like a behavior change, it is actually a bug fix. Currently, with `enable-api-fields` set to `stable`,  PipelineRuns like [this one](https://github.com/tektoncd/pipeline/blob/main/examples/v1/pipelineruns/beta/git-resolver.yaml) fail because the controller cannot create child TaskRuns. This change will result in a validation failure instead and this `pipelineSpec` will be prohibited with ``enable-api-fields=stable`.
```